### PR TITLE
Added Azure.ContainerApp.DisableAffinity

### LINKED ISSUE #2188: Session affinity may be preferable for stateful workloads hosted in Azure Container Apps
### Existing rule

_No response_

### Suggested rule

Azure Container Apps now supports session affinity, also known as sticky sessions.

This feature is in public preview.

Session affinity enables you to route all requests from a single client to the same Container Apps replica. This is useful for stateful workloads that require session affinity.

Session affinity is only supported when the app is in [single revision mode](https://learn.microsoft.com/nb-no/azure/container-apps/revisions#single-revision-mode) and the ingress type is HTTP.

Default behaviour should be off as stateless is the general recommendation, but customers can modify this by configuration in PSRule by enabling the rule.

### Pillar

Performance Efficiency

### Additional context

https://learn.microsoft.com/nb-no/azure/container-apps/sticky-sessions
https://learn.microsoft.com/en-us/azure/templates/microsoft.app/containerapps#ingress

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2188 

Added Azure.ContainerApp.DisableAffinity rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/en/rules/Azure.ContainerApp.DisableAffinity.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+# Disable session affinity
+
+## SYNOPSIS
+
+Disable session affinity to prevent unbalanced distribution.
+
+## DESCRIPTION
+
+Container apps allows you to configure session affinity (sticky sessions). It is feature that allows you to route all requests from a client to the same replica.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ContainerApp.DisableAffinity.md:17]:
```suggestion
Container apps allows you to configure session affinity (sticky sessions).
When enabled, this feature route requests from the same client to the same replica.
```

File: docs/en/rules/Azure.ContainerApp.DisableAffinity.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+This feature might be useful for stateful applications that require a consistent connection to the same replica. However, if your application does not store large amounts of state or cached data in memory (stateless application design pattern), session affinity might decrease your throughput because one replica could get overloaded with requests, while others are dormant.
+
+## RECOMMENDATION
+
+Consider disabling session affinity to prevent unbalanced distribution.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ContainerApp.DisableAffinity.md:23]:
```suggestion
Consider disabling session affinity to evenly distribute requests across each replica.
```

File: docs/en/rules/Azure.ContainerApp.DisableAffinity.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## EXAMPLES
+
+### Configure with Azure template
+
+To deploy Container Apps that pass this rule:
+
+- Set `properties.configuration.ingress.stickySessions.affinity` to `none` or dont specify the property at all.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ContainerApp.DisableAffinity.md:31]:
```suggestion
- Set `properties.configuration.ingress.stickySessions.affinity` to `none` or don't specify the property at all.
```

File: docs/en/rules/Azure.ContainerApp.DisableAffinity.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  }
+}
+```
+
+### Configure with Bicep
+
+To deploy Container Apps that pass this rule:
+
+- Set `properties.configuration.ingress.stickySessions.affinity` to `none` or dont specify the property at all.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ContainerApp.DisableAffinity.md:67]:
```suggestion
- Set `properties.configuration.ingress.stickySessions.affinity` to `none` or don't specify the property at all.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset. Just a few minor tweaks.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Thanks, I have updated the rule now.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**